server/src/actors/server/message/leave_lobby.rs

```diff
@@ -0,0 +1,57 @@
+use actix::{Handler, Message};
+use log::{error, info};
+use uuid::Uuid;
+
+use super::super::super::player::message as player_msg;
+use super::super::Server;
+
+#[derive(Message)]
+#[rtype(result = "()")]
+pub struct LeaveLobby {
+    pub player_id: Uuid,
+}
+
+impl Handler<LeaveLobby> for Server {
+    type Result = ();
+
+    fn handle(&mut self, msg: LeaveLobby, _ctx: &mut Self::Context) -> Self::Result {
+        let player_info = match self.players.get_mut(&msg.player_id) {
+            Some(player_info) => player_info,
+            None => {
+                error!("Player session '{}' not found", msg.player_id);
+                return;
+            }
+        };
+        if player_info.current_lobby.is_none() {
+            info!(
+                "Player '{}' is not in a lobby. Skipping LeaveLobby message",
+                player_info.username
+            );
+            return;
+        };
+        let lobby_id = player_info.current_lobby.unwrap();
+        let lobby_info = self.lobbies.get_mut(&lobby_id).unwrap();
+
+        player_info.current_lobby = None;
+        if lobby_info.players.len() > 1 {
+            lobby_info.players = lobby_info
+                .players
+                .iter()
+                .filter(|player| **player != msg.player_id)
+                .map(|player| *player)
```

Comment:
not sure if it will help but try `filter_map`...or maybe the fish operator `.collect::<Player>()` (I feel like that we should be able to avoid that `*player`)